src/Microsoft.CodeQuality.Analyzers.Exp/Core/FlowAnalysis/Framework/DataFlow/AnalysisEntityDataFlowOperationVisitor.cs

```diff
@@ -0,0 +1,269 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis.Operations.DataFlow.NullAnalysis;
+using Microsoft.CodeAnalysis.Operations.DataFlow.PointsToAnalysis;
+
+namespace Microsoft.CodeAnalysis.Operations.DataFlow
+{
+    /// <summary>
+    /// Operation visitor to flow the abstract dataflow analysis values for <see cref="AnalysisEntity"/> instances across a given statement in a basic block.
+    /// </summary>
+    internal abstract class AnalysisEntityDataFlowOperationVisitor<TAnalysisData, TAbstractAnalysisValue> : DataFlowOperationVisitor<TAnalysisData, TAbstractAnalysisValue>
```

Comment:
This class is mostly a refactoring - code related to `AnalysisEntity` is moved from the core `DataFlowOperationVisitor` to this sub-type to enable other non-AnalysisEntity based visitors, such as the `DisposeDataFlowOperationVisitor` which only needs to track data for (abstract) locations, not symbols.